lib/ci-helper.ts

```diff
@@ -559,6 +572,24 @@ export class CIHelper {
                 await addComment(`Submitted as [${
                     coverMid}](https://public-inbox.org/git/${coverMid})${
                         extraComment}`);
+            } else if (command === "/preview") {
+                if (argument && argument !== "") {
+                    throw new Error(`/preview does not accept arguments ('${
+                        argument}')`);
+                }
+
+                const pr = await getPRInfo(comment.prNumber);
+                const userInfo = await getUserInfo(comment.author);
+
+                if (!userInfo.email) {
+                    throw new Error(`Could not determine public email of ${
+                        comment.author}`);
+                }
+
+                const coverMid =
+                    await gitGitGadget.preview(pr, userInfo);
+                await addComment(`Preview email sent as [${
+                    coverMid}](https://public-inbox.org/git/${coverMid})`);
```

Comment:
Oh, completely forgot: do we want to provide a link here? Public-inbox won't have this mail, like, ever, because it was sent privately...

And now that I think of it, we should probably pay attention to _not_ Cc:ing the original authors during `/preview`. That could potentially be very irritating. I could imagine that `insertCcAndFromLines()` would need to be patched, but I might well be wrong.

Reply:
>  we should probably pay attention to _not_ Cc:ing the original authors during `/preview`.

I think we'll just need to guard [these lines](https://github.com/gitgitgadget/gitgitgadget/blob/bdcebdffd98610b6d9e295cf160b189d4e1d7746/lib/patch-series.ts#L600-L603) behind a new parameter `noCc: boolean` or something like it.

Reply:
1. Changed the comment to not be a markdown link.  It especially won't be valid if the message id is changed.
2. This is a preview so shouldn't all the names show up?  The provided send function in GitGitGadget::preview() removes the cc and changes the to for the email, so the envelope only has one to: address.

Reply:
>     1. Changed the comment to not be a markdown link.  It especially won't be valid if the message id is changed.

:+1:



>     2\. This is a preview so shouldn't all the names show up?  The provided send function in GitGitGadget::preview() removes the cc and changes the to for the email, so the envelope only has one to: address.

Oh, I missed that! That's a much more elegant solution than the one I proposed, I agree.